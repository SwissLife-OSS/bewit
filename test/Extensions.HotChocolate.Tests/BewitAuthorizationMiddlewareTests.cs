using System;
using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace Bewit.Extensions.HotChocolate.Tests
{
    public class BewitAuthorizationMiddlewareTests
    {
        [Fact]
        public async Task Query_WhenAuthorize_Success()
        {
            // arrange
            IServiceProvider serviceProvider = TestHelpers.CreateSchema<string>();
            var payload = "foo@bar.gmail.com";
            var token = await TestHelpers.CreateToken(serviceProvider, payload);

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(serviceProvider, token);

            // assert
            result.ToJson().MatchSnapshot();
        }

        [Fact]
        public async Task Query_WhenWrongToken_Fail()
        {
            // arrange
            IServiceProvider serviceProvider = TestHelpers.CreateSchema<object>();
            var token = await TestHelpers.CreateBadToken();

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(serviceProvider, token);

            // assert
            result.ToJson().MatchSnapshot();
        }

        [Fact]
        public async Task Query_WhenNotAuthorize_Fail()
        {
            // arrange
            IServiceProvider serviceProvider = TestHelpers.CreateSchema<object>();

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(serviceProvider);

            // assert
            result.ToJson().MatchSnapshot();
        }
    }
}
