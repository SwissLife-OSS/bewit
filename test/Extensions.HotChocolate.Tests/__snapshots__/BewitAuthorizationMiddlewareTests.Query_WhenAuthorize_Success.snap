{
  "Kind": "SingleResult",
  "Label": null,
  "Path": null,
  "Data": {
    "foo": "bar"
  },
  "Items": null,
  "Errors": null,
  "Extensions": null,
  "Incremental": null,
  "ContextData": null,
  "HasNext": null
}
